src/pages/Lend/Filter/FilterRangeSlider.vue

```diff
@@ -119,112 +166,158 @@ export default {
 
 				// eslint-disable-next-line no-param-reassign
 				range.max = 1000;
-
-			// minimum value
-			} else if (selectedRangeValue === range.min) {
-				if (this.minimumValueLabel === false) {
-					this.rangeLabel = this.minimumLabel;
-				} else {
-					this.rangeLabel = `${event.target.value}${this.minimumLabel}`;
-				}
-
-			// values in between min/max value
-			} else {
-				this.rangeLabel = `${event.target.value}${this.inbetweenLabel}`;
 			}
 
-			this.rangeValue = selectedRangeValue;
+			// this.rangeValue = selectedRangeValue;
 			let sliderPositionPercent = '100%';
 
 			if (this.minimumMaximumSwap === true) {
-				if (minimumValue === null && selectedRangeValue === null) {
-					minimumValue = range.max;
-					selectedRangeValue = range.max;
+				if (minimumValue === 0 && selectedRangeValue === 0) {
+					minimumValue = -1;
+					selectedRangeValue = 1000;
+
+					// eslint-disable-next-line no-param-reassign
+					range.min = -1;
+
+					// eslint-disable-next-line no-param-reassign
+					range.max = 1000;
 				} else {
-					minimumValue = selectedRangeValue;
-					selectedRangeValue = range.max;
+					minimumValue = this.selectedRangeValue;
+					selectedRangeValue = this.maximum;
 				}
 
 				sliderPositionPercent = numeral(minimumValue / selectedRangeValue).format('0%');
 			} else if (selectedRangeValue !== null) {
-				// eslint-disable-next-line max-len
-				sliderPositionPercent = numeral(Math.abs(selectedRangeValue - minimumValue) / Math.abs(range.max - minimumValue)).format('0%');
+				//	eslint-disable-next-line max-len
+				sliderPositionPercent = numeral(Math.abs(selectedRangeValue - minimumValue) / Math.abs(this.maximum - minimumValue)).format('0%');
 			}
 
-			// set colors of upper/lower range slider
-			// eslint-disable-next-line max-len
-			this.sliderStyle = `background: linear-gradient(to right, #4faf4e 0%, #4faf4e ${sliderPositionPercent}, #dbdbdb ${sliderPositionPercent}, #dbdbdb 100%)`;
+			this.setSlideStyle(sliderPositionPercent);
 
 			refine({ min: minimumValue, max: selectedRangeValue });
+
+			// set label of slider based on current position
+			this.selectedRangeValue = parseInt(event.target.value, 10);
+		},
+
+		// set colors of upper/lower range slider
+		setSlideStyle(sliderPositionPercent) {
+			// eslint-disable-next-line max-len
+			this.sliderStyle = `background-image: -webkit-linear-gradient(left, #4faf4e 0%, #4faf4e ${sliderPositionPercent}, #dbdbdb ${sliderPositionPercent}, #dbdbdb 100%);`;
 		},
 	},
 	created() {
 		if (this.minimumMaximumSwap === true) {
-			this.rangeLabel = this.minimumLabel;
-			this.sliderStyle = 'background: #dbdbdb';
+			this.selectedRangeValue = this.range.min;
+
+			// set slider to all gray
+			// eslint-disable-next-line max-len
+			this.sliderStyle = 'background-image: -webkit-linear-gradient(left, #4faf4e 0%, #4faf4e 0%, #dbdbdb 0%, #dbdbdb 100%)';
```

Comment:
Let's chat about `this.sliderStyle` tomorrow, it would be interesting to learn about the challenges in making this a computed property. Will also be interesting to understand why this code differs from `setSlideStyle`.